ReSwiftTests/StoreMiddlewareTests.swift

```diff
@@ -99,7 +101,10 @@ class StoreMiddlewareTests: XCTestCase {
         let reducer = TestValueStringReducer()
         let store = Store<TestStringAppState>(reducer: reducer.handleAction,
             state: TestStringAppState(),
-            middleware: [firstMiddleware, secondMiddleware, dispatchingMiddleware])
+            middleware: [
+                firstMiddleware as Middleware<TestStringAppState>,
+                secondMiddleware as Middleware<TestStringAppState>,
+                dispatchingMiddleware as Middleware<TestStringAppState>])
```

Comment:
I'm confused as to why the compiler is requiring this. `firstMiddleware` etc is `Middleware<StateType>`, and `TestStringAppState: StateType`.
Even doing this works:
```swift
let middleware: [Middleware<TestStringAppState>] = [
            firstMiddleware,
            secondMiddleware
        ]
        let store = Store<TestStringAppState>(reducer: reducer.handleAction,
            state: TestStringAppState(),
            middleware: middleware)
```

I think this is a swift bug. This should be compiling, as in `store.init`, `middleware: [Middleware<TestStringAppState>]`.

I'm going to verify the bug and submit it to Swift.
In the meantime, I think the pre-assignment I've provided here is a better solution than `as`

Reply:
Good, will change it. Can you tell me about the Swift ticket number? I'll then add a comment to it in both relevant places.

Reply:
https://bugs.swift.org/browse/SR-7362

Reply:
That was quick :)